salt/orchestrate/edx/cloudfront.sls

```diff
@@ -0,0 +1,113 @@
+
+{% for purpose, settings in env_settings.purposes.items() %}
+{% if 'live' in purpose %}
+{% set purpose_prefix = purpose.strip('-live').strip('-draft') %}
+{% set name = [purpose_prefix, ENVIRONMENT, 'cdn']|join('-') %}
+{% set purpose = distribution.pop('purpose', env_settings.purpose) %}
+{% set business_unit = distribution.pop('business_unit', BUSINESS_UNIT) %}
+provision_cloudfront_distribution_for_{{ purpose }}_in_{{ ENVIRONMENT }}:
+  boto_cloudfront.present:
+    - name: {{ name }}
+    - tags:
+        Environment: {{ ENVIRONMENT }}
+        Name: {{ name }}
+        business_unit: {{ business_unit }}
+        Department: {{ business_unit }}
+        OU: {{ business_unit }}
+        purpose: {{ purpose }}
+    - config:
+        Enabled: true
+        Aliases:
+          Quantity: 0
+        CacheBehaviors:
+          Quantity: 0
+        CallerReference: '1528233024979'
+        Comment: ''
+        CustomErrorResponses:
+          Quantity: 0
+        DefaultCacheBehavior:
+          AllowedMethods:
+            CachedMethods:
+              Items:
+              - HEAD
+              - GET
+              Quantity: 2
+            Items:
+            - HEAD
+            - GET
+            - OPTIONS
+            Quantity: 3
+          Compress: true
+          DefaultTTL: 86400
+          ForwardedValues:
+            Cookies:
+              Forward: none
+            Headers:
+              Items:
+              - Host
+              - Origin
+              - Referer
+              Quantity: 3
+            QueryString: false
+            QueryStringCacheKeys:
+              Quantity: 0
+          LambdaFunctionAssociations:
+            Quantity: 0
+          MaxTTL: 31536000
+          MinTTL: 0
+          SmoothStreaming: false
+          TargetOriginId: Custom-{{ settings.domains.lms }}
+          TrustedSigners:
+            Enabled: false
+            Quantity: 0
+          ViewerProtocolPolicy: allow-all
+        DefaultRootObject: ''
+        HttpVersion: http2
+        IsIPV6Enabled: true
+        Logging:
+          Bucket: ''
+          Enabled: false
+          IncludeCookies: false
+          Prefix: ''
+        Origins:
+          Items:
+          {% for suffix in ['live', 'draft'] %}
+          {% for domain in ['lms', 'cms'] %}
+          - CustomHeaders:
+              Quantity: 0
+            CustomOriginConfig:
+              HTTPPort: 80
+              HTTPSPort: 443
+              OriginKeepaliveTimeout: 5
+              OriginProtocolPolicy: match-viewer
+              OriginReadTimeout: 30
+              OriginSslProtocols:
+                Items:
+                - TLSv1.1
+                - TLSv1.2
+                Quantity: 3
+            DomainName: {{ env_settings.purposes[purpose_prefix + suffix].domains[domain] }}
+            Id: Custom-{{ env_settings.purposes[purpose_prefix + suffix].domains[domain] }}
+            OriginPath: ''
+          {% endfor %}
+          {% endfor %}
+          Quantity: 4
```

Comment:
Shouldn't this be 2?

Reply:
That's 4 because I added origins for LMS and CMS on both draft and live